scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/Crypto.scala

```diff
@@ -16,25 +16,108 @@
 
 package com.intel.analytics.bigdl.ppml.crypto
 
+import com.intel.analytics.bigdl.dllib.nn.NormMode.Value
 import com.intel.analytics.bigdl.ppml.utils.Supportive
+import org.apache.spark.input.PortableDataStream
+
+import java.io.{DataInputStream, DataOutputStream}
+import javax.crypto.Cipher
 
 trait Crypto extends Supportive with Serializable {
-  def encryptFile(sourceFilePath: String, saveFilePath: String, dataKeyPlaintext: String)
-  def decryptFile(sourceFilePath: String, saveFilePath: String, dataKeyPlaintext: String)
-  def encryptBytes(sourceBytes: Array[Byte], dataKeyPlaintext: String): Array[Byte]
-  def decryptBytes(sourceBytes: Array[Byte], dataKeyPlaintext: String): Array[Byte]
-}
-
-object CryptoMode extends Enumeration {
-  type CryptoMode = Value
-  val PLAIN_TEXT = value("plain_text", "plain_text")
-  val AES_CBC_PKCS5PADDING = value("AES/CBC/PKCS5Padding", "AES/CBC/PKCS5Padding")
-  val UNKNOWN = value("UNKNOWN", "UNKNOWN")
-  class EncryptModeEnumVal(name: String, val value: String) extends Val(nextId, name)
-  protected final def value(name: String, value: String): EncryptModeEnumVal = {
-    new EncryptModeEnumVal(name, value)
-  }
-  def parse(s: String): Value = {
-    values.find(_.toString.toLowerCase() == s.toLowerCase).getOrElse(CryptoMode.UNKNOWN)
+  def init(cryptoMode: CryptoMode, mode: OperationMode, dataKeyPlaintext: String): Unit
+
+  def decryptBigContent(ite: Iterator[(String, PortableDataStream)]): Iterator[String]
+
+  def genFileHeader(): Array[Byte]
+
+  def verifyFileHeader(header: Array[Byte]): Unit
+
+  def update(content: Array[Byte]): Array[Byte]
+
+  def update(content: Array[Byte], offset: Int, len: Int): Array[Byte]
+
+  def doFinal(content: Array[Byte]): (Array[Byte], Array[Byte])
+
+  def doFinal(content: Array[Byte], offset: Int, len: Int): (Array[Byte], Array[Byte])
+
+  def encryptStream(inputStream: DataInputStream, outputStream: DataOutputStream): Unit
+
+  def decryptStream(inputStream: DataInputStream, outputStream: DataOutputStream): Unit
+
+  def decryptFile(binaryFilePath: String, savePath: String): Unit
+
+  def encryptFile(binaryFilePath: String, savePath: String): Unit
+}
+
+object Crypto {
+  def apply(cryptoMode: CryptoMode): Crypto = {
+    cryptoMode match {
+      case AES_CBC_PKCS5PADDING =>
+        new BigDLEncrypt()
+      case default =>
+        throw new EncryptRuntimeException("No such crypto mode!")
```

Comment:
do not throw exception here; use the new error API (https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/doc/error-log-api.md). @dding3 - have we enforced the style check for scala?

Reply:
> do not throw exception here; use the new error API (https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/doc/error-log-api.md). @dding3 - have we enforced the style check for scala?

Type mismatch, if I just change the this line to `Log4Error.invalidInputError(true, "Unsupported crypto mode!")`
I need some work around.

Reply:
> I just change the this line to `Log4Error.invalidInputError(true, "Unsupported crypto mode!")`

We have enforced no assert, no require check in scala, disallow `throw exception` is on the way.

Reply:
> > do not throw exception here; use the new error API (https://github.com/intel-analytics/BigDL/blob/main/docs/readthedocs/source/doc/error-log-api.md). @dding3 - have we enforced the style check for scala?
> 
> Type mismatch, if I just change the this line to `Log4Error.invalidInputError(true, "Unsupported crypto mode!")` I need some work around.

I think after call `Log4Error.invalidInputError(true, "Unsupported crypto mode!")`, you can add `null` to fix the compile issue